synapse/storage/databases/main/push_rule.py

```diff
@@ -839,21 +812,27 @@ async def copy_push_rule_from_room_to_room(
             rule: A push rule.
         """
         # Create new rule id
-        rule_id_scope = "/".join(rule["rule_id"].split("/")[:-1])
+        rule_id_scope = "/".join(rule.rule_id.split("/")[:-1])
         new_rule_id = rule_id_scope + "/" + new_room_id
 
+        new_conditions = []
+
         # Change room id in each condition
-        for condition in rule.get("conditions", []):
+        for condition in rule.conditions:
+            new_condition = condition
             if condition.get("key") == "room_id":
-                condition["pattern"] = new_room_id
+                new_condition = dict(condition)
```

Comment:
Is taking a copy of the condition here a bugfix?

Reply:
Oh, err, maybe? I thought we copied the push rule by this point but I can't find where we do that now